File: android/src/main/java/dev/barooni/capacitor/calendar/models/results/GetDefaultCalendarResult.kt
package dev.barooni.capacitor.calendar.models.results

import com.getcapacitor.JSObject
import dev.barooni.capacitor.calendar.models.data.CalendarInfo
import dev.barooni.capacitor.calendar.models.templates.JSResult

data class GetDefaultCalendarResult(
    private val calendar: CalendarInfo?,
) : JSResult {
    override fun toJSON(): JSObject =
        JSObject().apply {
            put(
                "result",
                calendar?.let { cal ->
                    JSObject().apply {
                        put("id", cal.id)
                        put("title", cal.title)
                        put("internalTitle", cal.internalName)
                        put("color", cal.color)
                        put("isImmutable", null)
                        put("allowsContentModifications", null)
                        put("type", null)
                        put("isSubscribed", null)
                        put("source", null)
                        put("visible", cal.visible)
                        put("accountName", cal.accountName)
                        put("ownerAccount", cal.ownerAccount)
                        put("maxReminders", cal.maxReminders)
                        put("location", cal.location)
                    }
                },
            )
        }
}

File: ios/Sources/CapacitorCalendarPlugin/Models/Results/GetDefaultCalendarResult.swift
import EventKit
import Capacitor

struct GetDefaultCalendarResult: JSResult {
    let calendar: EKCalendar?

    func toJSON() -> JSObject {
        var result = JSObject()
        if let calendar = calendar {
            result["result"] = ImplementationHelper.calendarToJSObject(calendar)
        } else {
            result["result"] = NSNull()
        }
        return result
    }
}
